engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -1046,17 +1046,14 @@ static int CreateTextureAsync(lua_State* L)
     CreateTextureResourceParams create_params = {};
     CheckCreateTextureResourceParams(L, &create_params);
 
-    // Create an empty upload buffer
-    dmBuffer::HBuffer upload_buffer = create_params.m_Buffer;
-    bool use_upload_buffer = create_params.m_Buffer == 0;
-    if (use_upload_buffer)
+    // We need to create an empty upload buffer if no explicit buffer is passed
+    bool is_transcoded              = create_params.m_CompressionType != dmGraphics::TextureImage::COMPRESSION_TYPE_DEFAULT && dmGraphics::IsFormatTranscoded(create_params.m_CompressionType);
```

Comment:
Is the "!= dmGraphics::TextureImage::COMPRESSION_TYPE_DEFAULT" needed?
Seems the "IsFormatTranscoded(type)" already should handle this question?